sentry_sdk/hub.py

```diff
@@ -441,38 +441,138 @@ def start_span(
     ):
         # type: (...) -> Span
         """
-        Create a new span whose parent span is the currently active
-        span, if any. The return value is the span object that can
-        be used as a context manager to start and stop timing.
-
-        Note that you will not see any span that is not contained
-        within a transaction. Create a transaction with
-        ``start_span(transaction="my transaction")`` if an
-        integration doesn't already do this for you.
+        Create and start timing a new span whose parent is the currently active
+        span or transaction, if any. The return value is a span instance,
+        typically used as a context manager to start and stop timing in a `with`
+        block.
+
+        Only spans contained in a transaction are sent to Sentry. Most
+        integrations start a transaction at the appropriate time, for example
+        for every incoming HTTP request. Use `start_transaction` to start a new
+        transaction when one is not already in progress.
         """
-
-        client, scope = self._stack[-1]
+        # XXX: if the only way to use start_span correctly is when there
+        # is already an existing transaction/span in the scope, then
+        # this is a hard to use API. We don't document nor support
+        # appending an existing span to a new transaction created to
+        # contain the span.
+
+        # TODO: consider removing this in a future release.
+        # This is for backwards compatibility with releases before
+        # start_transaction existed, to allow for a smoother transition.
+        if isinstance(span, Transaction) or "transaction" in kwargs:
+            deprecation_msg = (
```

Comment:
I don't think we need the redundant check here for `isinstance(span, Transaction) or "transaction" in kwargs`, we should just extract `deprecation_msg` to be a top level constant

Reply:
The intention here was to have a self-contained block with the TODO note, so it is clearer what it applies to.